src/stream/stream/cycle.rs

```diff
@@ -42,7 +42,7 @@ where
                 this.source.set(this.orig.clone());
                 this.source.poll_next(cx)
             }
-            item => Poll::Ready(item),
+            item @ Some(..) => Poll::Ready(item),
```

Comment:
Unnecessary

Reply:
I think the goal here is to make clearer to readers that the only possibility is `Some(..)`, rather than being unclear about the possibly multiple cases it covers.